### ISSUE TITLE
Complete rewrite in Kotlin

### ISSUE THREAD

**Issue by babariviere** [babariviere/flutter_sms#23]
This rewrite will make error handling better because of Kotlin features

**Comment by joanpablo** [comment]
That's really cool

**Comment by babariviere** [comment]
Yeah but I take too much time to do it haha

**Comment by babariviere** [comment]
You are completely right, but for now it's experimental. As I said in another issue, I want to finish the rewrite in kotlin to implement it properly
